### PR TITLE
Pia 3328 error screen topbar

### PR DESCRIPTION
**Pull request by chriskryn**
Top and bottom navigation bar

### CODE REVIEW

File: BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift [hunk 1]
```diff
@@ -280,22 +280,20 @@ extension GiniBankNetworkingScreenApiCoordinator {
...
-                
-                networkDelegate.displayError(withMessage: .localized(resource: AnalysisStrings.analysisErrorMessage),
-                                             andAction: {
-                                                 self.startAnalysisWithReturnAssistant(networkDelegate: networkDelegate)
-                                             })
-                 
+                DispatchQueue.main.async { [weak self] in
+                    guard error != .requestCancelled else { return }
+                    self?.displayError(errorType: ErrorType(error: error))
+
```
Comment (vizaknaidavid) [inline BankSDK/GiniBankSDK/Sources/GiniBankSDK/Core/GiniBankNetworkingScreenApiCoordinator.swift:286]: [Suggestion] No need for empty line here

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorScreenViewController.swift [hunk 2]
```diff
@@ -35,6 +30,17 @@ class ErrorScreenViewController: UIViewController {
             secondTitle: NSLocalizedStringPreferredFormat(
                 "ginicapture.error.backToCamera",
                 comment: "Back to camera"))
+        view.translatesAutoresizingMaskIntoConstraints = false
+        if viewModel.isEnterManuallyHidden() == false {
```
Comment (vizaknaidavid) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorScreenViewController.swift:34]: `if viewModel.isEnterManuallyHidden() == false {
             view.enterButton.isHidden = false
         } else {
             view.enterButton.isHidden = true
         }`
         
         This whole block could be written like this:
         
         `view.enterButton.isHidden = viewModel.isEnterManuallyHidden()`
         
         Same for the one below

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                }
+            } else {
+                self = .serverError
+            }
+        default:
+            self = .unexpected
+        }
+    }
+
+    func iconName() -> String {
```
Comment (vizaknaidavid) [inline CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:47]: [Suggestion] Declaring these as variables not functions would reduce complexity

Comment (vizaknaidavid) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:47]: `
var iconName: String {

        switch self {
        case .connection:
            return "errorCloud"
        case .request:
            return "alertTriangle"
        case .authentication:
            return "errorAuth"
        case .serverError:
            return "errorGlobe"
        case .unexpected:
            return "alertTriangle"
        case .uploadIssue:
            return "errorUpload"
        case .importError:
            return "alertTriangle"
        }
    }`

**Review by vizaknaidavid — CHANGES_REQUESTED** [review]
Looks good, I just left some tiny suggestions

File: CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift (hunk 3 above)
Comment (chriskryn) [reply CaptureSDK/GiniCaptureSDK/Sources/GiniCaptureSDK/Core/Screens/Error/ErrorType.swift:47]: hmm @vizaknaidavid  What kind of complexity ? I see that the difference would be just the way of invoking it without '()' parenthesis.

**Comment by chriskryn** [comment]
Merged into PIA-3333 https://github.com/gini/gini-mobile-ios/pull/203

**Pull request closed without merging**